### ISSUE TITLE
Swipe gesture to dismiss ImageViewer is broken 👇💥

### ISSUE THREAD

**Issue by renatorodrigues** [MailOnline/ImageViewer#26]
In the _single image mode_, the vertical swipe gesture to dismiss the image away only works properly when swiping up.  When swiping down, the gesture never finishes, thus, never dismissing the controller.

![image-viewer](https://cloud.githubusercontent.com/assets/246079/14459087/c92d7116-00ad-11e6-857b-fb763af2f931.gif)

**Comment by Krisiacik** [comment]
Hi @renatorodrigues I checked the issue and it would be sufficient just to add this line
`presentingViewController?.dismissViewControllerAnimated(true, completion: nil)`
 to the else case.
But @RuiAAPeres suggested to extract this to a separate method. After examining your fix with abs, I realised you also removed a bit of duplication so I hope you don't mind us taking a hint from your approach and issue a fix in-house instead.

Thanks for help! 👍
